src/ray/raylet/node_manager.cc

```diff
@@ -409,7 +409,9 @@ void NodeManager::Heartbeat() {
   auto heartbeat_data = std::make_shared<HeartbeatTableData>();
   heartbeat_data->set_node_id(self_node_id_.Binary());
   RAY_CHECK_OK(
-      gcs_client_->Nodes().AsyncReportHeartbeat(heartbeat_data, /*done*/ nullptr));
+      gcs_client_->Nodes().AsyncReportHeartbeat(heartbeat_data, [](Status status) {
+        RAY_CHECK(status.ok()) << "This node has beem marked as dead.";
```

Comment:
```suggestion
        if (status.disconnected()) {
             RAY_LOG(FATAL) << "your string"
        }
```
I think this is safer because the current logic can kill raylet because of other status errors?